src/main.rs

```diff
@@ -23,32 +23,37 @@ struct Cli {
     #[clap(parse(from_str = parse_dependency))]
     dependencies: Vec<Dependency>,
 
+    /// Create a library instead of a binary.
+    #[clap(short, long)]
+    lib: bool,
```

Comment:
```suggestion
    /// Create a library instead of a binary.
    #[clap(long)]
    lib: bool,
```

My idea is that the parameters must look like the parameters of cargo. In cargo:
```
        --bin                      Use a binary (application) template [default]
        --lib                      Use a library template
```
As you can see, there is no `-l` to make a library.

You can keep the `-l` if you want but I don't recommend it.

Reply:
This one was just for stay consistent about the help page

```plain
cargo-temp 
This tool allow you to create a new Rust temporary project in a temporary directory.

The dependencies can be provided in arguments (e.g. `cargo-temp anyhow tokio`). When the shell is
exited, the temporary directory is deleted unless you removed the file `TO_DELETE`

USAGE:
    cargo-temp [FLAGS] [OPTIONS] [--] [dependencies]...

ARGS:
    <dependencies>...
            Dependencies to add to `Cargo.toml`.
            
            The default version used is `*` but this can be replaced using `=`. E.g. `cargo-temp
            anyhow=1.0.13`

FLAGS:
    -h, --help
            Prints help information

    -l, --lib
            Create a library instead of a binary

    -V, --version
            Prints version information


OPTIONS:
        --name <project-name>
            Name of the temporary crate

        --worktree <worktree-branch>...
            Create a git working tree
```

Reply:
The help page is generated based on your struct. I'm not sure I understand your argument.